### ISSUE TITLE
Joyent Cloud Stack tracing in 2016.3

### ISSUE THREAD

**Issue by Ch3LL** [saltstack/salt#33402]
### Description of Issue/Question

When attempting to create a VM in 2016.3.0 with joyent cloud provider it stack traces and does not create the vm. When running `salt-cloud -p joyent-test ch3ll-joyent` I receive this stack trace:

```
name: ch3ll-joyent
None
[15:35:12,413][salt.cloud.cli                           :442 ][ERROR   ] There was a profile error: 'NoneType' object has no attribute '__getitem__'
Traceback (most recent call last):
  File "/testing/salt/cloud/cli.py", line 284, in run
    self.config.get('names')
  File "/testing/salt/cloud/__init__.py", line 1446, in run_profile
    ret[name] = self.create(vm_)
  File "/testing/salt/cloud/__init__.py", line 1281, in create
    output = self.clouds[func](vm_)
  File "/testing/salt/cloud/clouds/joyent.py", line 318, in create
    query_instance(vm_)
  File "/testing/salt/cloud/clouds/joyent.py", line 220, in query_instance
    'wait_for_ip_interval_multiplier', vm_, __opts__, default=1),
  File "/testing/salt/utils/cloud.py", line 2315, in wait_for_ip
    data = update_callback(*update_args, **update_kwargs)
  File "/testing/salt/cloud/clouds/joyent.py", line 191, in _query_ip_address
    data = show_instance(vm_['name'], call='action')
  File "/testing/salt/cloud/clouds/joyent.py", line 653, in show_instance
    ret = query(command='my/machines/{0}'.format(node['id']),
TypeError: 'NoneType' object has no attribute '__getitem__'
```

### Setup
Here is my joyent profile config:

```
joyent-test:
  provider: joyent-config
  size: Extra Small 512 MB
  image: ubuntu-certified-14.04
  location: us-east-1
  script_args: '-P -Z'
```

### Steps to Reproduce Issue
1. Add the above profile config 
2. `salt-cloud -p joyent-test ch3ll-test`

### Versions Report
```
[root@li983-166 cloud-test-configs]# salt --versions-report
Salt Version:
           Salt: 2016.3.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: centos 7.0.1406 Core
        machine: x86_64
        release: 4.1.5-x86_64-linode61
         system: Linux
```

**Comment by ssgward** [comment]
@techhat - we'd like to get this fixed for 2016.3 if possible.

**Comment by Ch3LL** [comment]
I'm going to go ahead and close this. I tested the fix and its now working
